tools/feature_version_audit.py

```
@@ -0,0 +1,685 @@
+        for fname, info in sorted(feature_actions.items()):
+            author = f" ({info['author']})" if info['author'] else ""
+            lines.append(f"- **{fname}**{author}: " + "; ".join(info["actions"]))
+    return lines
+
+def format_feature_table(feature_analysis):
+    lines = []
+    lines.append("| Feature | Prior Ver | Proposed Ver | Needs Bump | Change Types | Note | Commit |")
+    lines.append("|---------|-----------|--------------|------------|--------------|------|--------|")
+    for fa in feature_analysis:
+        def bold(val):
+            return f"**{val}**" if fa['is_attention'] and val != '' and val != '-' else val
```

Ruff (0.11.9)

Function definition does not bind loop variable `fa`

(B023)